src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,67 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int[] output = readFile(fromFileName);
+        String result = getResult(output);
+        writeFile(result, toFileName);
+    }
+
+    public void writeFile(String result, String toFileName) {
+        BufferedWriter writer = null;
+            try {
+                writer = new BufferedWriter(new FileWriter(toFileName, true));
+                writer.write(result);
+            } catch (IOException e) {
+                throw new RuntimeException("Can't write data to file", e);
+            } finally {
+                try {
+                    writer.close();
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+
+    }
+
+    public int[] readFile(String fromFileName) {
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(fromFileName));
+            StringBuilder stringBuilder = new StringBuilder();
+            String value = reader.readLine();
+            while (value != null) {
+                stringBuilder.append(value).append(System.lineSeparator());
+                value = reader.readLine();
+            }
+            String string = stringBuilder.toString();
+            String[] split = string.split(System.lineSeparator());
+            int supply = 0;
+            int buy = 0;
+            for (int i = 0; i < split.length; i++) {
+                String substring = split[i].substring(split[i].indexOf(",") + 1);
+                int parseInt = Integer.parseInt(substring);
+                if (split[i].substring(0, split[i].indexOf(",")).equals("supply")) {
+                    supply += parseInt;
+                } else {
+                    buy += parseInt;
+                }
+            }
+            return new int[] {supply, buy, (supply - buy)};
```

Comment:
The variables 'supply' and 'buy' should be declared as local variables inside the readFile method instead of being class fields if they are not present.